agent/connect/ca/provider.go

```diff
@@ -7,6 +7,36 @@ import (
 
 //go:generate mockery -name Provider -inpkg
 
+// ProviderConfig encapsulates all the data Consul passes to `Configure` on a
+// new provider instance. The provider must treat this as read-only and make
+// copies of any map or slice if it might modify them internally.
+type ProviderConfig struct {
+	// ClusterID is the current Consul cluster ID.
+	ClusterID string
+
+	// Datacenter is the current Consul datacenter.
+	Datacenter string
+
+	// PrimaryDC is true when the CA instance is in the primary DC typically it
+	// may choose to act as a root in this case while secondaries are typically
+	// intermediate CAs. In some case the primary DC in Consul is an intermediate
+	// signed by some external CA along with that CA's public cert so the old name
+	// of `IsRoot` was misleading.
+	PrimaryDC bool
```

Comment:
Maybe (for consistency) call this `PrimaryDatacenter` or `IsPrimaryDatacenter` since the companion field `Datacenter` isn't abbreviated.